x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportStartDataFrameTransformAction.java

```diff
@@ -249,7 +267,15 @@ public boolean test(PersistentTasksCustomMetaData.PersistentTask<?> persistentTa
                 return true;
             }
             // We just want it assigned so we can tell it to start working
-            return assignment != null && assignment.isAssigned();
+            return assignment != null && assignment.isAssigned() && isNotStopped(persistentTask);
+        }
+
+        // checking for `isNotStopped` as the state COULD be marked as failed for any number of reasons
+        // But if it is in a failed state, _stats will show as much and give good reason to the user.
+        // If it is not able to be assigned to a node all together, we should just close the task completely
+        private boolean isNotStopped(PersistentTasksCustomMetaData.PersistentTask<?> task) {
+            DataFrameTransformState state = (DataFrameTransformState)task.getState();
+            return state != null && state.getTaskState().equals(DataFrameTransformTaskState.STOPPED) == false;
```

Comment:
Since I am now starting the indexer in the node executor class, a new task should move away from the state of `STOPPED` eventually, after being allocated. Of course, if it is not allocated, that will never happen (hence keeping the allocation checks). 

As for only verifying that the task is NOT stopped, setting it to a failed state is a real possibility and since the executor has no direct means of notifying us, setting the task to failed with a reason is good enough.